CRM/Event/BAO/Event.php

```diff
@@ -1446,11 +1454,9 @@ public static function displayProfile(&$params, $gid, &$groupTitle, &$values, &$
         $fields = CRM_Core_BAO_UFGroup::getFields($gid, FALSE, CRM_Core_Action::ADD);
       }
 
-      foreach ($fields as $v) {
-        if (!empty($v['groupTitle'])) {
-          $groupTitle['groupTitle'] = $v['groupTitle'];
-          break;
-        }
+      if (!empty($fields)) {
+        $profile = civicrm_api3('UFGroup', 'getsingle', ['id' => $gid, ['return' => ['title', 'frontend_title']]]);
```

Comment:
@eileenmcnaughton any reason not to use `CRM_UF_BAO_UFGroup::getFrontEndTitle()` here?

Reply:
because it didn't exist when I wrote those lines....

Reply:
I think it would probably be good to tidy this up before merging but otherwise looks good to me

Reply:
ok done